app/models/course_user.rb

```diff
@@ -35,6 +35,14 @@ def experience_points
 
   delegate :experience_points, to: :experience_points_records
 
+  # Computes the level number of the CourseUser with respect to
+  # a course's Course::Levels.
+  #
+  # @return [Fixnum] CourseUser level number
+  def level_number
```

Comment:
I didn't notice, but you should put instance methods after class instance methods (the one that goes `def self.something`)